x-pack/plugins/security_solution/public/common/containers/cases/use_get_related_cases_by_event.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback, useState, useEffect } from 'react';
+import { useKibana, useToasts } from '../../lib/kibana';
+import { CASES_ERROR_TOAST } from '../../components/event_details/insights/translations';
+import { APP_ID } from '../../../../common/constants';
+
+type RelatedCases = Array<{ id: string; title: string }>;
+
+export const useGetRelatedCasesByEvent = (eventId: string) => {
+  const {
+    services: { cases },
+  } = useKibana();
+  const toasts = useToasts();
+
+  const [relatedCases, setRelatedCases] = useState<RelatedCases | undefined>(undefined);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<null | unknown>(null);
+
+  const getRelatedCases = useCallback(async () => {
+    setLoading(true);
+    let relatedCasesResponse: RelatedCases = [];
+    try {
+      if (eventId) {
+        relatedCasesResponse =
+          (await cases.api.getRelatedCases(eventId, {
+            owner: APP_ID,
+          })) ?? [];
+      }
+    } catch (err) {
+      setError(err);
+      toasts.addWarning(CASES_ERROR_TOAST(err));
+    } finally {
+      setLoading(false);
+    }
+    setRelatedCases(relatedCasesResponse);
```

Comment:
Consider moving this statement into the `finally` block if `setRelatedCases` should always be set to the value of `relatedCasesResponse` (even if an error were thrown from within the `catch` block)

Reply:
Will be in the next push, Thanks!

Reply:
Will be in the next push!